Forte.EpiServer.AzureSearch/Plugin/ContentIndexingScheduledJob.cs

```diff
@@ -48,21 +51,32 @@ public override string Execute()
                 VisitedContent = new HashSet<ContentReference>(),
                 Statistics = new IndexStatistics(),
             };
-            
+
             OnStatusChanged("Indexing content start...");
             _contentIndexer.Index(ContentReference.RootPage, indexContentRequest).GetAwaiter().GetResult();
 
+            var exceptionsThresholdReached = indexContentRequest.Statistics.Exceptions.Count >= indexContentRequest.ExceptionThreshold;
+
+            if (exceptionsThresholdReached)
+            {
+                message += "Exceptions threshold reached. " +
+                           $"Exceptions: {string.Join(NewLine + NewLine, indexContentRequest.Statistics.Exceptions.Select(e => e.ToString()))}{NewLine}{NewLine}" +
+                           $"Failed for ids: {string.Join(",", indexContentRequest.Statistics.FailedIds.Select(c => c.ID))}";
+
+                throw new ScheduledJobFailedException(message);
+            }
+
             if (!_cancellationToken.IsCancellationRequested)
             {
                 OnStatusChanged("Clearing outdated items...");
-                _indexGarbageCollector.RemoveOutdatedContent(jobStartTime).GetAwaiter().GetResult();                
+
+                var contentIdsThatCanStayOutdated = indexContentRequest.Statistics.FailedIds.Select(c => c.ID).ToList();
```

Comment:
once you change `ContentIdString` to `ContentComplexReference` you'll need to change this to `Select(c => c.ToString())`. Maybe it's also worth considering if `RemoveOutdatedContent` should accept `IList<ContentReference>` instead of `IList<int>`

This change, in general, has one more advantage: if we would use only ID part (int) we're missing information about ContentProviders - there might be two different documents with ID == 10 but these will differ in `ContentReference.ProviderName` property